src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -343,11 +343,15 @@ internal static Diagnostic Create(DiagnosticInfo info)
                 return null;
             }
 
+            if (compilation.SemanticModelProvider == null)
+            {
+                compilation = compilation.WithSemanticModelProvider(new CachingSemanticModelProvider());
```

Comment:
>new CachingSemanticModelProvider() [](start = 68, length = 34)

What are we trying to achieve by this and why? Consider explaining this in a comment.

Reply:
I will revert this change - I originally thought this was a hot path, but this is actually a separate public API that is not used during analyzer execution.